src/client.rs

```diff
@@ -837,4 +837,24 @@ mod tests {
         let body_should_be = serde_json::to_string(&json_data).unwrap();
         assert_eq!(buf, body_should_be);
     }
+
+    #[test]
+    fn add_json_fail() {
+        use serde::{Serialize, Serializer};
+        use serde::ser::Error;
+        struct MyStruct;
+        impl Serialize for MyStruct {
+            fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+                where S: Serializer
+                {
+                    Err(S::Error::custom("nope"))
+                }
+        }
+
+        let client = Client::new().unwrap();
+        let some_url = "https://google.com/";
+        let r = client.post(some_url);
+        let json_data = MyStruct{};
+        assert_eq!(format!("{}", r.json(&json_data).unwrap_err()), "nope".to_string());
```

Comment:
I would just check the error kind here rather than relying on the format:

```rust
assert!(r.json(&bad_json).unwrap_err().is_serialization());
```